src/main/java/com/atlassian/db/replica/internal/SqlFunction.java

```diff
@@ -0,0 +1,443 @@
+            "exists",
+            "in",
+            "any",
+            "some",
+            "all",
+            "generate_series",
+            "generate_subscripts",
+            "current_catalog",
+            "current_database",
+            "current_query",
+            "current_role",
+            "current_schema",
+            "current_schemas",
+            "current_user",
+            "inet_client_addr",
+            "inet_client_port",
+            "inet_server_addr",
+            "inet_server_port",
+            "pg_backend_pid",
+            "pg_conf_load_time",
+            "pg_is_other_temp_schema",
+            "pg_listening_channels",
+            "pg_my_temp_schema",
+            "pg_postmaster_start_time",
+            "pg_trigger_depth",
+            "session_user",
+            "user",
+            "version",
+            "has_any_column_privilege",
+            "has_column_privilege",
+            "has_database_privilege",
+            "has_foreign_data_wrapper_privilege",
+            "has_function_privilege",
+            "has_language_privilege",
+            "has_schema_privilege",
+            "has_sequence_privilege",
+            "has_server_privilege",
+            "has_table_privilege",
+            "has_tablespace_privilege",
+            "has_type_privilege",
+            "pg_has_role",
+            "pg_collation_is_visible",
+            "pg_conversion_is_visible",
+            "pg_function_is_visible",
+            "pg_opclass_is_visible",
+            "pg_operator_is_visible",
+            "pg_opfamily_is_visible",
+            "pg_table_is_visible",
+            "pg_ts_config_is_visible",
+            "pg_ts_dict_is_visible",
+            "pg_ts_parser_is_visible",
+            "pg_ts_template_is_visible",
+            "pg_type_is_visible",
+            "format_type",
+            "pg_describe_object",
+            "pg_identify_object",
+            "pg_get_constraintdef",
+            "pg_get_expr",
+            "pg_get_functiondef",
+            "pg_get_function_arguments",
+            "pg_get_function_identity_arguments",
+            "pg_get_function_result",
+            "pg_get_indexdef",
+            "pg_get_keywords",
+            "pg_get_ruledef",
+            "pg_get_serial_sequence",
+            "pg_get_triggerdef",
+            "pg_get_userbyid",
+            "pg_get_viewdef",
+            "pg_options_to_table",
+            "pg_tablespace_databases",
+            "pg_tablespace_location",
+            "pg_typeof",
+            "collation",
+            "to_regclass",
+            "to_regproc",
+            "to_regprocedure",
+            "to_regoper",
+            "to_regoperator",
+            "to_regtype",
+            "col_description",
+            "shobj_description",
+            "txid_current",
+            "txid_current_snapshot",
+            "txid_snapshot_xip",
+            "txid_snapshot_xmax",
+            "txid_snapshot_xmin",
+            "txid_visible_in_snapshot"
```

Comment:
@wyrzyk How we would discover the missing function here? other than manual 1:1 analysis, once someone discovers that sth should be faster.
As a systematic way, I'd imagine getting a list of unique calls done by Jira within 1 workweek day and weekend and filter functions to find out potential missed.

Though when I write about it I already started thinking that this is over-engineering.

Reply:
We could emit a `new RouteDecision(Reason.FUNCTION_CALL, functionName)`. We'd be abusing the `String sql` field, but it could be generalized.

Reply:
@afaruga-atlassian, @dagguh Thanks for the feedback. The diagnostics API is still in progress. We currently see we miss API method name for `Reason.RW_API_CALL`, and functionName for `Reason.FUNCTION_CALL`. I think about adding it to be more generic (3rd use case would help here).

Reply:
3rd use case: SQL for `Reason.WRITE_OPERATION` (due to `isUpdate(sql) || isDelete(sql)`).
Do we need the SQL for other `Reason`s?